src/screen-cast.c

```diff
@@ -624,12 +915,98 @@ collect_screen_cast_stream_data (GVariantIter *streams_iter)
   return streams;
 }
 
+static void
+generate_and_save_restore_token (ScreenCastSession *screen_cast_session)
+{
+  Session *session = (Session *)screen_cast_session;
+
+  if (!screen_cast_session->restore_data)
+    {
+      g_clear_pointer (&screen_cast_session->restore_token, g_free);
+      g_clear_pointer (&screen_cast_session->restore_data, g_variant_unref);
+      return;
+    }
+
+  switch (screen_cast_session->persist_mode)
+    {
+    case PERSIST_MODE_NONE:
+      g_clear_pointer (&screen_cast_session->restore_token, g_free);
+      g_clear_pointer (&screen_cast_session->restore_data, g_variant_unref);
```

Comment:
Should the transient/persintent permission be cleared here too, if there existed a token?

Same question applies to transient when setting persistent, and persistent when setting transient.

Reply:
Yup. Done